CPTSuperLearn/interpolator.py

```diff
@@ -24,6 +26,33 @@ def predict(self, prediction_points: List[float]):
         raise NotImplementedError("Predict method must be implemented")
 
 
+class SchemaGANInterpolator(InterpolatorAbc):
+    def __init__(self, model_path: str):
+        self.model_path = model_path
+        self.model = load_model(model_path)
+        self.size_x = 512

```

Comment:
create a check to see if the size of the images being interpolated is the same as the size of the schemaGAN model. raise error if not

Reply:
ok!